test/cqlpy/test_map_key_selection.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+#############################################################################
+# Tests for ...
+#############################################################################
+
+import pytest
+import re
+import time
+from cassandra.protocol import InvalidRequest
+from .util import unique_name, unique_key_string
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<int, int>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<text, int>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+@pytest.fixture(scope="module")
+def table3(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int PRIMARY KEY, m map<text, frozen<map<text, set<int>>>>)")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+def test_basic_int_key_selection(cql, table1):
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (0, {1:10,2:20})")
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (1, {1:11,2:21,3:31})")
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (2, {3:32})")
+    assert set(cql.execute(f"SELECT p,m[1] FROM {table1}")) == {(0,10),(1,11),(2,None)}
+    assert set(cql.execute(f"SELECT p,m[2] FROM {table1}")) == {(0,20),(1,21),(2,None)}
+    assert set(cql.execute(f"SELECT p,m[3] FROM {table1}")) == {(0,None),(1,31),(2,32)}
+
+def test_basic_string_key_selection(cql, table2):
+    cql.execute("INSERT INTO " + table2 + "(p,m) VALUES (0, {'aa':10,'ab':20})")
+    list(cql.execute(f"SELECT m['aa'] FROM {table2} WHERE p=0")) == [(10,)]
+    list(cql.execute(f"SELECT m['ab'] FROM {table2} WHERE p=0")) == [(20,)]
+    list(cql.execute(f"SELECT m['ac'] FROM {table2} WHERE p=0")) == [(None,)]
+
+def test_recursive_key_selection(cql, table3):
+    cql.execute("INSERT INTO " + table3 + "(p, m) VALUES (0, {'1': {'a': {1, 2, 4}, 'b': {3}}, '2': {'a': {2, 4}}})")
+    assert list(cql.execute(f"SELECT m['1']['a'] FROM {table3} WHERE p=0")) == [({1,2,4},)]
+    assert [set(r.m1) for r in cql.execute(f"SELECT m['1']['b'] AS m1 FROM {table3} WHERE p=0")] == [{3}]
+    assert [set(r.m2) for r in cql.execute(f"SELECT m['2']['a'] AS m2 FROM {table3} WHERE p=0")] == [{2,4}]
+    assert [r.m2 for r in cql.execute(f"SELECT m['2']['b'] AS m2 FROM {table3} WHERE p=0")] == [None]
+
+def test_prepare_key(cql, table1):
+    cql.execute("INSERT INTO " + table1 + "(p,m) VALUES (0, {1:10,2:20})")
+    with pytest.raises(InvalidRequest, match="Unsupported bind"):
+        cql.prepare(f"SELECT m[?] FROM {table1} WHERE p = ?")
```

Comment:
Why is this not supported? Is it also not supported in Cassandra?
If it's not supported by accident - not deliberately - then we will need to mark this test "xfail" and explain the situation. Or better yet, fix the code to support it and not fail - i.e., it should be able to do what Cassandfra can do.

Reply:
I added support of this in a new commit